src/graphforge/datasets/loaders/compression.py

```
@@ -0,0 +1,113 @@
+    # Open zstd-compressed file
+    with archive_path.open("rb") as compressed_file:
+        # Create zstd decompressor
+        dctx = zstd.ZstdDecompressor()
+
+        # Decompress and extract tar
+        with (
+            dctx.stream_reader(compressed_file) as reader,
+            tarfile.open(fileobj=reader, mode="r|") as tar,
+        ):
+            # Extract all files
+            tar.extractall(extract_to)
```

Ruff (0.14.14)

[error] Uses of `tarfile.extractall()`

(S202)